### PR TITLE
Document new only expunge deletes behavior for Force Merge API

### LINKED ISSUE #5388: [DOC] Document force_merge API (and new `only_expunge_deletes` behavior in 2.12)
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** 
Relates to https://github.com/opensearch-project/OpenSearch/pull/10603, which doesn't change any APIs but does change existing behavior. 

I don't believe we currently have any documentation around the `force_merge` API. We do have https://opensearch.org/docs/latest/dashboards/im-dashboards/forcemerge/, but that's just a UI component to simplify access to the `force_merge` API. I think we should have a section for `force_merge` under https://opensearch.org/docs/latest/api-reference/index-apis/index/.

Anyway, this Stackoverflow question (https://stackoverflow.com/questions/55903641/is-it-ok-to-forcemerge-with-only-expunge-deletes-on-a-read-write-index) from 2019 essentially covers the problem solved by https://github.com/opensearch-project/OpenSearch/pull/10603. 

Essentially, under normal circumstances, OpenSearch will do background segment merges that produce segments with size at most `index.merge.policy.max_merged_segment` (default 5GB). When you force merge, that max segment size is ignored -- every shard could potentially get merged down to a single segment, regardless of size. The problem is that you now have segments bigger than 5GB that may accumulate deleted documents. Those segments cannot possibly participate in any future merges unless the total deletion count would bring them back under 5GB.

Given that there's a parameter called `only_expunge_deletes`, it's not unreasonable that some people (including me -- I've been burned by this) would think that it's safe to use that parameter to force a merge that only expunges deletes (without producing giant zombie segments that never participate in future merges). Unfortunately, until now, those people would have been wrong.

This change makes it so that `only_expunge_deletes` will still honor the value of `index.merge.policy.max_merged_segment`.

**What other resources are available?**
N/A

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Closes #5388 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@msfroh Could you please review this PR when you get a chance? Thanks!

**Review by msfroh — APPROVED** [review]
Looks great! Thanks @kolchfa-aws

File: _api-reference/index-apis/force-merge.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ The force merge API operation forces a merge on the shards of one or more indexe
 
 In OpenSearch, a shard is a Lucene index, which consists of _segments_ (or segment files). Segments store the indexed data. Periodically, smaller segments are merged into larger ones and the larger segments become immutable. Merging reduces the overall number of segments on each shard and frees up disk space. 
 
+OpenSearch performs background segment merges that produce segments whose size equals to at most `index.merge.policy.max_merged_segment` (the default is 5 GB).
```
Comment (msfroh) [inline _api-reference/index-apis/force-merge.md:18]: Maybe reword to "produce segments whose size is at most" ?

Comment (kolchfa-aws) [reply _api-reference/index-apis/force-merge.md:18]: This is better, thanks!

Comment (kolchfa-aws) [inline _api-reference/index-apis/force-merge.md:18]:
```suggestion
OpenSearch performs background segment merges that produce segments whose size is at most `index.merge.policy.max_merged_segment` (the default is 5 GB).
```

File: _api-reference/index-apis/force-merge.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ The force merge API operation forces a merge on the shards of one or more indexe
 
 In OpenSearch, a shard is a Lucene index, which consists of _segments_ (or segment files). Segments store the indexed data. Periodically, smaller segments are merged into larger ones and the larger segments become immutable. Merging reduces the overall number of segments on each shard and frees up disk space. 
 
+OpenSearch performs background segment merges that produce segments whose size is at most `index.merge.policy.max_merged_segment` (the default is 5 GB).
```
Comment (natebower) [inline _api-reference/index-apis/force-merge.md:18]:
```suggestion
OpenSearch performs background segment merges that produce segments no larger than `index.merge.policy.max_merged_segment` (the default is 5 GB).
```

**Review by natebower — APPROVED** [review]
LGTM with one change.
